src/main/java/com/github/pgasync/impl/netty/NettyPgProtocolStream.java

```diff
@@ -115,10 +123,25 @@ public NettyPgProtocolStream(EventLoopGroup group, SocketAddress address, boolea
 
             pushSubscriber(subscriber);
             write(messages);
+            readTimeoutScheduled = initReadTimeout(subscriber);
 
         }).lift(throwErrorResponsesOnComplete());
     }
 
+    private ScheduledFuture<?> initReadTimeout(Subscriber subscriber) {
+        if (readTimeout != 0) {
+            Runnable onTimeout = () -> {
+                ReadTimeoutException timeoutException = ReadTimeoutException.INSTANCE;
+                subscriber.onError(timeoutException);
+                ctx.fireExceptionCaught(timeoutException);
+                ctx.close();
```

Comment:
Closing this channel requires connection removal from the poll. Validator needs to be updated (it is in one of other PRs).